File: skypydb-js/src/utils/embedding_callable.ts
import type { EmbeddingMatrix } from "../types";

/**
 * Base callable interface for embedding providers.
 */
export abstract class EmbeddingCallable {
  abstract embed(texts: string[]): Promise<EmbeddingMatrix>;

  async call(texts: string[]): Promise<EmbeddingMatrix> {
    return this.embed(texts);
  }
}

/**
 * Backward compatible alias kept for existing internal imports.
 */
export abstract class EmbeddingCallableMixin extends EmbeddingCallable {}

File: skypydb/utils/embedding_callable.py
"""Mixin that allows embedding providers to be used as callables."""

from typing import List


class EmbeddingCallable:
    """Protocol-like helper that makes embedding providers callable."""

    def __call__(
        self,
        texts: List[str]
    ) -> List[List[float]]:
        """Make embedding providers callable for compatibility with integrations."""

        return self.embed(texts)


# Deprecated compatibility alias. Prefer `EmbeddingCallable`.
EmbeddingCallableMixin = EmbeddingCallable
